packages/mux-player-react/src/index.tsx

```diff
@@ -194,9 +194,26 @@ const usePlayer = (
       return defaultHasChanged(playerEl, value, propName);
     }
   );
-  // NOTE: Somewhere in the codebase, `currentTime` is getting cast to a number, resulting in `NaN` + an error.
-  // This is a bandaid solution for now. (CJP)
-  useObjectPropEffect('currentTime', currentTime ?? 0, ref);
+  // NOTE: Need to handle currentTime differently to manage
+  // 1. Whether the prop value is/was "unset" to e.g. `undefined`
+  //    - To make sure we don't try to update currentTime on these changes
+  // 2. Whether the *prop* value was changed ("from the outside")
+  //    - To only apply the `currentTime` if it actually changed from the last provided value
+  // (CJP)
+  const [internalCurrentTime, setInternalCurrentTime] = useState(currentTime);
+  useEffect(() => {
+    if (internalCurrentTime == currentTime) return;
+    setInternalCurrentTime(currentTime);
+  }, [currentTime, internalCurrentTime]);
```

Comment:
```suggestion
  }, [currentTime, internalCurrentTime]);
  
```

a newline here would be nice. was difficult to read